src/util.rs

```diff
@@ -281,9 +281,16 @@ pub mod utf8 {
 	use libc::c_char;
 
 	pub fn into<'a>(data: *const c_char, length: u32) -> Vec<&'a str> {
-		unsafe {
```

Comment:
@mjkillough oh yeah, I agree about returning early when `length == 0`, what I meant was about the rest of the code, what you're essentially doing is popping the last element of the resulting `Vec`, but it's harder to follow and easier to get wrong in future changes, so I'd still go with keeping the old code but popping the last element if it is empty for the actual splitting part.

The early return when `length == 0` is good. I'm not sure I'm explaining myself (tired after a flight) so I'll give it another shot tomorrow if I'm being cryptic 🐼